lib/ide/src/controller/module.rs

```diff
@@ -20,10 +21,64 @@ use data::text::TextChangedNotification;
 use file_manager_client as fmc;
 use json_rpc::error::RpcError;
 use parser::api::IsParser;
+use parser::api::ModuleWithMetadata;
 use parser::Parser;
+
+use serde::Serialize;
+use serde::Deserialize;
 use shapely::shared;
 
 
+
+// ============
+// == Module ==
+// ============
+
+/// Parsed file / module with metadata
+#[derive(Debug,Clone)]
+pub struct Module {
+    /// ast representation
+    pub ast: Ast,
+    /// strongly typed metadata
+    pub metadata: Metadata
+}
```

Comment:
Ok something is very broken here. You have `ModuleWithMetadata` and `Module`, which constains ... `metadata`.